hatchery/pods.go

```diff
@@ -79,35 +92,49 @@ func statusK8sPod(userName string) (*WorkspaceStatus, error) {
 		status.Status = "Stopped"
 		return &status, nil
 	case "Pending":
-		status.Status = "Launching"
-		return &status, nil
 	case "Running":
-		break
-	default:
-		fmt.Printf("Unknown pod status for %s: %s\n", podName, string(pod.Status.Phase))
-	}
+		var allReady = true
+		if pod.Status.Phase == "Pending" {
+			allReady = false
+		} else {
+			for _, v := range pod.Status.Conditions {
+				if v.Type == "Ready" {
+					if v.Status != "True" {
+						allReady = false
+					}
+				} else if v.Type == "PodScheduled" {
+					if v.Status != "True" {
+						allReady = false
+					}
+				}
+			}
+		}
 
-	var allReady = true
-	for _, v := range pod.Status.Conditions {
-		if v.Type == "Ready" {
-			if v.Status != "True" {
-				allReady = false
+		if allReady == true {
+			status.Status = "Running"
+			return &status, nil
+		} else {
+			status.Status = "Launching"
+			conditions := make([]PodConditions, len(pod.Status.Conditions))
+			for i, cond := range pod.Status.Conditions {
+				conditions[i].Status = string(cond.Status)
+				conditions[i].Type = string(cond.Type)
 			}
-		} else if v.Type == "PodScheduled" {
-			if v.Status != "True" {
-				allReady = false
+			status.Conditions = conditions
+			containerStates := make([]ContainerStates, len(pod.Status.ContainerStatuses))
+			for i, cs := range pod.Status.ContainerStatuses {
+				containerStates[i].State = cs.State
+				containerStates[i].Name = cs.Name
+				containerStates[i].Ready = cs.Ready
 			}
+			status.ContainerStates = containerStates
+			return &status, nil
```

Comment:
You could move `return &status, nil` after the conditional to avoid needing that code both above and here. Or just remove it all together from every place in the switch since we already have a `return` statement after the switch